CONTRIBUTORS.md

```diff
@@ -35,6 +35,30 @@ This project and everyone participating in it is governed by this [Code of Condu
 - Any major changes should be added to the [CHANGELOG](CHANGELOG.md).
 - Mention any required documentation changes in the description of your pull request.
 - If adding an RPC endpoint, add an entry for the new endpoint to the OpenAPI spec `./docs/rpc/openapi.yaml`.
+- If your code adds or modifies any major features (struct, trait, test, module, etc.), each should be 
+  documented according to the [Rust style guide](https://doc.rust-lang.org/rustdoc/how-to-write-documentation.html) 
+
+### Documentation Style 
+We will try to follow the [Rust style guide](https://doc.rust-lang.org/rustdoc/how-to-write-documentation.html). 
+Here's a contrived example of code following the guide: 
+```rust
+/// Returns 0 if the input is less than or equal to 5, and panics otherwise.
+/// 
+/// # Panics 
+/// The function panics if the input is greater than 5.
+pub fn example(i: int) -> int {
```

Comment:
Should we be describing the inputs?

Reply:
From [the guide](https://doc.rust-lang.org/rustdoc/how-to-write-documentation.html#documenting-components):

> Because the type system does a good job of defining what types a function passes and returns, there is no benefit of explicitly writing it into the documentation, especially since rustdoc adds hyper links to all types in the function signature.

Reply:
I agree we shouldn't document things that are redundant in the context, and the type declarations can save a lot of comments. In some cases, though, there remains ambiguity even given the type.

Here are some examples, from the rust code base, of methods without comments that I believe are ambiguous without comments:

## Example 1

What is "input"?
```
    fn compute_cost(
        &mut self,
        cost_function: ClarityCostFunction,
        input: &[u64],
    ) -> Result<ExecutionCost, CostErrors> {
```

## Example 2
Height of what? Burn chain or Stacks chain?
```
    fn get_burn_header_hash(
        &self,
        height: u32,
        sortition_id: &SortitionId,
    ) -> Option<BurnchainHeaderHash>;
```

# Example 3
What is `mblock_header_1` vs `mblock_header_2`?
```
    pub fn run_poison_microblock(
        &mut self,
        sender: &PrincipalData,
        mblock_header_1: &StacksMicroblockHeader,
        mblock_header_2: &StacksMicroblockHeader,
    ) -> Result<Value, Error> {
```